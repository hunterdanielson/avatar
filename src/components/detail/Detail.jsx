import React from 'react';
import PropTypes from 'prop-types';

const Detail = ({ name, photoUrl, affiliation, allies, enemies }) => (
  <div>
    <h1>{name}</h1>
    <img src={photoUrl} />
    <h2>{affiliation}</h2>
    <h3>Allies</h3>
    <ul>
      {allies.map(ally => <li key={ally}>{ally}</li>)}
    </ul>
    <h3>Enemies</h3>
    <ul>
      {enemies.map(enemy => <li key={enemy}>{enemy}</li>)}
    </ul>
  </div>

);

Detail.propTypes = {
  name: PropTypes.string.isRequired,
  photoUrl: PropTypes.string.isRequired,
  affiliation: PropTypes.string.isRequired,
  allies: PropTypes.arrayOf(PropTypes.string),
  enemies: PropTypes.arrayOf(PropTypes.string)
};

export default Detail;


