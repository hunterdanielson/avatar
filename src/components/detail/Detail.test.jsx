import React from 'react';
import { shallow } from 'enzyme';
import Detail from './Detail';

describe('Detail component', () => {
  it('renders Detail', () => {
    const wrapper = shallow(<Detail  name='Aang' photoUrl='pic.png' affiliation='avatar' enemies={['azula', 'ozai']} allies={['katara']}/>);
    expect(wrapper).toMatchSnapshot();
  });
});
