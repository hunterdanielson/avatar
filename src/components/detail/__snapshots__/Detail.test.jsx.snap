// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Detail component renders Detail 1`] = `
<div>
  <h1>
    Aang
  </h1>
  <img
    src="pic.png"
  />
  <h2>
    avatar
  </h2>
  <h3>
    Allies
  </h3>
  <ul>
    <li
      key="katara"
    >
      katara
    </li>
  </ul>
  <h3>
    Enemies
  </h3>
  <ul>
    <li
      key="azula"
    >
      azula
    </li>
    <li
      key="ozai"
    >
      ozai
    </li>
  </ul>
</div>
`;
